src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp

```diff
@@ -666,3 +692,72 @@ void runtime::intelgpu::do_logic_kernel(cldnn::topology& topology,
                                                  {1});
     topology.add(op_logical);
 }
+
+void runtime::intelgpu::do_reverse_operation(cldnn::topology& topology,
+                                             const string& input_name,
+                                             const Shape& input_shape,
+                                             const string& output_name,
+                                             const Shape& output_shape,
+                                             const element::Type& output_type,
+                                             const AxisSet& reversed_axes)
+{
+    const cldnn::layout layout = IntelGPULayout::create_cldnn_layout(output_type, output_shape);
+    const string entry_point_name = "reverse_" + output_name;
+    codegen::CodeWriter writer;
+    std::vector<size_t> gws;
+
+    writer << "__kernel void " << entry_point_name << "(const __global float input"
+           << array_dims(input_shape) << ", __global float output" << array_dims(output_shape)
+           << ")\n";
+
+    writer.block_begin();
+    {
+        size_t var_idx = 0;
+        // Current clDNN implementation supports only 3 dimension of global work sizes
+        for (auto const& i : output_shape)
```

Comment:
Could you please make a `static vector<size_t> generate_loops_begin()` to generate this sequence of the initial loops?